RetrospectiveExtension.Frontend/components/feedbackBoardMetadataFormPermissions.tsx

```diff
@@ -0,0 +1,207 @@
+
+    return orderedPermissionOptions;
+  }
+
+  const emitChangeEvent = () => {
+    props.onPermissionChanged({
+      permissions: {
+        Teams: teamPermissions,
+        Members: memberPermissions
+      }
+    })
+  }
+
+  const PermissionImage = (props: {option: FeedbackBoardPermissionOption}) => {
+    if (props.option.type === 'team') {
+      return <i className="permission-image fa-solid fa-users h-11 w-11"></i>
+    }
+
+    return <img className="permission-image" src={props.option.thumbnailUrl} alt={`Permission image for ${props.option.name}`} />
+  }
+
+  const PublicWarningBanner = () => {
+    if(teamPermissions.length === 0 && memberPermissions.length === 0) {
+      return <div className="board-metadata-form-section-information">
+        <i className="fas fa-exclamation-circle" aria-label="Board is visible to every member in the organization"></i>&nbsp;This board is visible to every member in the organization.
+      </div>
+    }
+
+    return null;
+  }
+
+  useEffect(() => {
+    setSelectAllState();
+    setFilteredPermissionOptions(orderedPermissionOptions(filteredPermissionOptions));
+    emitChangeEvent();
+  }, [teamPermissions, memberPermissions])
+
+  return <div className="board-metadata-form board-metadata-form-permissions">
+    <section className="board-metadata-form-board-settings board-metadata-form-board-settings--no-padding">
+      <PublicWarningBanner />
+      
+      <div className="search-bar">
+        <TextField
+          ariaLabel="Search for a team or a member to add permissions"
+          aria-required={true}
+          placeholder={'Search teams or users'}
+          id="retrospective-permission-search-input"
+          value={searchTerm}
+          onChange={(_, newValue) => handleSearchTermChanged(newValue)}
+        />
+      </div>
+
+      <div className="board-metadata-table-container">
+        {/* TODO: Replace with Fluent grid once we migration components over */}
+        <table className="board-metadata-table">
+          <thead>
+            <tr>
+              <th className="cell-checkbox">
+                <Checkbox
+                  className="my-2"
+                  id="select-all-permission-options-visible"
+                  ariaLabel="Select all permission options that are visible in the permission option list."
+                  boxSide="start"
+                  checked={selectAllChecked}
+                  onChange={(_, checked) => handleSelectAllClicked(checked)}
+                />
+              </th>
+              <th className={"text-left"}>
+                <span aria-label="Permission option name table header">{"Name"}</span>
+              </th>
+            </tr>
+          </thead>
+          <tbody>
+            {filteredPermissionOptions.map((option, index) => {
+              return (
+                <tr key={'table-row-' + index} className="option-row">
+                  <td>
+                    <Checkbox
+                      className="my-2"
```

Comment:
is this class-name the final choice?

Reply:
that's coming from tailwind :)

and it means `margin in y direction is 2 units` 🤦‍♂️

perfectly fine in this case